### ISSUE TITLE
The default CIDRs for serviceSubnet and podSubnet should be documented.

### ISSUE THREAD

**Issue by victor-sudakov** [kubernetes-sigs/kind#3117]
**What would you like to be documented**:
I think that the default address pools for the serviceSubnet and podSubnet should be documented. Currently you can only learn them from studying the source code.

**Why is this needed**:
This is needed because some people use non-default address pools in the Docker daemon settings resulting in a possibility of Kind node addresses overlapping with the  serviceSubnet and podSubnet address pools which I was told kindnet does not detect and does not work with.

**Comment by aojea** [comment]
This is a good issue for new contributors, the values are defined here

https://github.com/kubernetes-sigs/kind/blob/77675009ce7862102f2d4a6c6b61b8094f964ebb/pkg/apis/config/v1alpha4/default.go#L51-L72

/help

**Comment by victor-sudakov** [comment]
BTW why do you have such an odd prefixlen for the IPv6  service subnet (longer than a /64)? `fd00:10:96::/112` looks kind of odd to me. A more appropriate ULA prefix length would be like /48 or /56.

**Comment by aojea** [comment]
> `fd00:10:96::/112` looks kind of odd to me. A more appropriate ULA prefix length would be like /48 or /56. Is there a specific reason such a long prefixlen has been chosen?

hehe, long history https://github.com/kubernetes/enhancements/pull/1534

TL;DR this is a limitation in the implementation of the allocator that tracks teh assigned IPs to Services, it has to use a bitmap that is snapshotted to etcd, if the bitmap is too large it is not possible to store it in etcd or the performance is super bad ... hopefully I'll be able to solve this problem soon in kubernetes

**Comment by Sajiyah-Salat** [comment]
I got the issue. we need to document podsubnet and servicesubnet in documentation which is ``10.96.0.0/16,  10.244.0.0/16``. Can you please specify the place where this would be useful to document.

**Comment by aojea** [comment]
This looks like the better place

https://kind.sigs.k8s.io/docs/user/configuration/#networking

**Comment by victor-sudakov** [comment]
> TL;DR this is a limitation in the implementation of the allocator that tracks teh assigned IPs to Services, it has to use a bitmap

OMG :-)

**Comment by BenTheElder** [comment]
Besides the KEP pointer there's a rather involved PR ongoing to replace that Kubernetes allocator, Antonio's actually been authoring that and I'm intermittently attempting to help review :-)

https://github.com/kubernetes/kubernetes/pull/115075

**Comment by DharanidharSai** [comment]
Hey I am new here how can I contribute and learn I need help I at present I can do non code contribution

**Comment by BenTheElder** [comment]
@DharanidharSai Please see https://kind.sigs.k8s.io/docs/contributing/getting-started/ which covers everything including how to contribute to the docs.

@victor-sudakov Heh - Antonio _just_ got the new allocator into alpha in 1.27 🎉 

Presumably kubeadm will still not change defaults though, and it's not clear if we can/should either, at least not until we're not supporting older Kubernetes prior to this hitting GA at some point and we have a new kind config API revision after that but ... some progress is better than no progress.

**Comment by aojea** [comment]
go ahead, send a PR

**Comment by RijulGulati** [comment]
@aojea raised PR for the same.

**Comment by RijulGulati** [comment]
I believe this is good to close? @aojea @BenTheElder

**Comment by aojea** [comment]
> I believe this is good to close? @aojea @BenTheElder

yeah
